### PR TITLE
feat: Add Scan Type and Options TCE-1026

### PR DESCRIPTION
**Pull request by mrfyda**

### CODE REVIEW

File: src/model/codacyInput.ts [hunk 1]
```diff
@@ -30,12 +30,16 @@ export class Tool {
   }
 }
 
+export type Options = { [k: string]: any }
```
Comment (codacy-vrhpires) [inline src/model/codacyInput.ts:33]: Should limit it to `string`,  `number` and `boolean`. Use `any` as little as possible.

File: src/model/specification.ts [hunk 2]
```diff
@@ -35,6 +35,18 @@ export type SecuritySubcategory =
   | "Regex"
   | "SSL"
 
+export type ScanType =
```
Comment (codacy-vrhpires) [inline src/model/specification.ts:38]: Should use an `enum` (and change the others enumerations while we're at it).

File: src/test/fileUtils.spec.ts [hunk 3]
```diff
@@ -156,19 +155,31 @@ describe("fileUtils", () => {
...
+            category: "CodeStyle",
+            parameters: [],
+            enabled: false
           }
         ]
       }
       deepStrictEqual(result, expected)
     })
   })
+  it("should parse a codacyrc file with options", () => {
```
Comment (codacy-vrhpires) [inline src/test/fileUtils.spec.ts:173]: To be coherent with the remaining tests, `options` should be added to the first test and here it should test without `options` set.

File: src/model/codacyInput.ts (hunk 1 above)
Comment (mrfyda) [reply src/model/codacyInput.ts:33]: according to the specification, Codacy can send any value here, not just `string`, `number` or `boolean`.

File: src/model/specification.ts (hunk 2 above)
Comment (mrfyda) [reply src/model/specification.ts:38]: is there an advantage to `enum` over the union types?

File: src/model/codacyInput.ts (hunk 1 above)
Comment (codacy-vrhpires) [reply src/model/codacyInput.ts:33]: Unexpected.

File: src/model/specification.ts (hunk 2 above)
Comment (codacy-vrhpires) [reply src/model/specification.ts:38]: There is an _tabs vs spaces_ kind of debate about those. I'll take types then. 😄

**Review by codacy-vrhpires — APPROVED** [review]

**Pull request merged**